HBASE-28811 Use region server configuration for evicting the cache wh…

ragarkar opened the pull request:
…ile unassigning a region

Apache-HBase commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 38s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  |  Patch does not have any anti-patterns.  |
|||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  3s |  |  master passed  |
| +1 :green_heart: |  compile  |   3m  0s |  |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 38s |  |  master passed  |
| +1 :green_heart: |  spotbugs  |   1m 37s |  |  master passed  |
| +1 :green_heart: |  spotless  |   0m 46s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 53s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   3m  2s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   3m  2s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -0 :warning: |  checkstyle  |   0m 36s | [/results-checkstyle-hbase-server.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6197/1/artifact/yetus-general-check/output/results-checkstyle-hbase-server.txt) |  hbase-server: The patch generated 4 new + 0 unchanged - 0 fixed = 4 total (was 0)  |
| +1 :green_heart: |  spotbugs  |   1m 39s |  |  the patch passed  |
| +1 :green_heart: |  hadoopcheck  |  10m 36s |  |  Patch does not cause any errors with Hadoop 3.3.6 3.4.0.  |
| -1 :x: |  spotless  |   0m 38s |  |  patch has 63 errors when running spotless:check, run spotless:apply to fix.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 12s |  |  The patch does not generate ASF License warnings.  |
|  |   |  35m 49s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6197/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6197 |
| Optional Tests | dupname asflicense javac spotbugs checkstyle codespell detsecrets compile hadoopcheck hbaseanti spotless |
| uname | Linux 35ed6fc47e5a 5.4.0-186-generic #206-Ubuntu SMP Fri Apr 26 12:31:10 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / c28816109e975b5140ec700249529ad245774648 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| spotless | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6197/1/artifact/yetus-general-check/output/patch-spotless.txt |
| Max. process+thread count | 83 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6197/1/console |
| versions | git=2.34.1 maven=3.9.8 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 41s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  4s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 59s |  |  master passed  |
| +1 :green_heart: |  compile  |   0m 59s |  |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 30s |  |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 22s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 55s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 58s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 58s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 29s |  |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 19s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 226m 23s |  |  hbase-server in the patch passed.  |
|  |   | 251m 28s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6197/1/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6197 |
| Optional Tests | javac javadoc unit compile shadedjars |
| uname | Linux 0c330eb4e6ca 5.4.0-186-generic #206-Ubuntu SMP Fri Apr 26 12:31:10 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / c28816109e975b5140ec700249529ad245774648 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6197/1/testReport/ |
| Max. process+thread count | 5956 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6197/1/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 38s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  |  Patch does not have any anti-patterns.  |
|||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 32s |  |  master passed  |
| +1 :green_heart: |  compile  |   4m  1s |  |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 45s |  |  master passed  |
| +1 :green_heart: |  spotbugs  |   1m 51s |  |  master passed  |
| +1 :green_heart: |  spotless  |   0m 49s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 37s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   3m 57s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   3m 57s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   0m 47s |  |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   1m 58s |  |  the patch passed  |
| +1 :green_heart: |  hadoopcheck  |  11m 41s |  |  Patch does not cause any errors with Hadoop 3.3.6 3.4.0.  |
| +1 :green_heart: |  spotless  |   0m 45s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 10s |  |  The patch does not generate ASF License warnings.  |
|  |   |  42m  8s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6197/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6197 |
| Optional Tests | dupname asflicense javac spotbugs checkstyle codespell detsecrets compile hadoopcheck hbaseanti spotless |
| uname | Linux e3bcc601def3 5.4.0-186-generic #206-Ubuntu SMP Fri Apr 26 12:31:10 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 016ed6562e9b313d94ac15c984a01cf535227394 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| Max. process+thread count | 84 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6197/2/console |
| versions | git=2.34.1 maven=3.9.8 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

wchevreuil commented on hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/TransitRegionStateProcedure.java:
```diff
@@ -372,14 +372,12 @@ private Flow confirmOpened(MasterProcedureEnv env, RegionStateNode regionNode)
   private void closeRegionAfterUpdatingMeta(MasterProcedureEnv env, RegionStateNode regionNode) {
-    CloseRegionProcedure closeProc =
+    LOG.debug("Close region: isSplit: {}: evictOnClose: {}", isSplit,
       isSplit
```
Nit: this log is misleading, we should just pass the `evictCache` valure for `evictOnClose`.

wchevreuil commented on hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/TransitRegionStateProcedure.java:
```diff
@@ -372,14 +372,12 @@ private Flow confirmOpened(MasterProcedureEnv env, RegionStateNode regionNode)
-          assignCandidate, evictCache);
+        ? env.getMasterConfiguration().getBoolean(EVICT_BLOCKS_ON_SPLIT_KEY, DEFAULT_EVICT_ON_SPLIT)
+        : evictCache);
+    CloseRegionProcedure closeProc = new CloseRegionProcedure(this, getRegion(),
```
Let's put a comment explaining this change:
`Splits/Merges are special cases, rather then deciding on the cache eviction behaviour here at Master, we just need to tell this close is for a split/merge and let RSes decide on the eviction. See HBASE-28811 for more context.`

wchevreuil reviewed (commented):
Can we add UT for the changed behaviour? Something similar to TestSplitWithCache, but start with a given value for evictOnClose, change it for RSes only, trigger a region close, check the cache:
1) Set evictOnClose to false:
`UTIL.getConfiguration().setBoolean(EVICT_BLOCKS_ON_CLOSE_KEY, false);`
2) Start cluster:
`UTIL.startMiniCluster(1);`
3) Change the config for the RS only:
`UTIL.getMiniHBaseCluster().getRegionServer(0).getConfiguration().setBoolean(EVICT_BLOCKS_ON_CLOSE_KEY, true);`
4) Disable table:
`UTIL.getAdmin().disableTable();`
5) Check blocks got evicted:

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 25s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  |  Patch does not have any anti-patterns.  |
|||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 15s |  |  master passed  |
| +1 :green_heart: |  compile  |   3m  6s |  |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 35s |  |  master passed  |
| +1 :green_heart: |  spotbugs  |   1m 35s |  |  master passed  |
| +1 :green_heart: |  spotless  |   0m 46s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  2s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   3m  6s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   3m  6s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   0m 36s |  |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   1m 40s |  |  the patch passed  |
| +1 :green_heart: |  hadoopcheck  |  11m 43s |  |  Patch does not cause any errors with Hadoop 3.3.6 3.4.0.  |
| +1 :green_heart: |  spotless  |   0m 43s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 10s |  |  The patch does not generate ASF License warnings.  |
|  |   |  37m 50s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6197/3/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6197 |
| Optional Tests | dupname asflicense javac spotbugs checkstyle codespell detsecrets compile hadoopcheck hbaseanti spotless |
| uname | Linux 20783683072d 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 578ca4da02e09b5c21d2d2b6a9235b1f99c8589f |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| Max. process+thread count | 85 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6197/3/console |
| versions | git=2.34.1 maven=3.9.8 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 40s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  1s |  |  master passed  |
| +1 :green_heart: |  compile  |   0m 59s |  |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 29s |  |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 23s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 59s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  0s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   1m  0s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 31s |  |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 23s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 231m 10s |  |  hbase-server in the patch passed.  |
|  |   | 256m  8s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6197/3/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6197 |
| Optional Tests | javac javadoc unit compile shadedjars |
| uname | Linux fa5d4ce7a853 5.4.0-192-generic #212-Ubuntu SMP Fri Jul 5 09:47:39 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 578ca4da02e09b5c21d2d2b6a9235b1f99c8589f |
| Default Java | Eclipse Adoptium-17.0.11+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6197/3/testReport/ |
| Max. process+thread count | 4906 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6197/3/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

ragarkar commented:
> Can we add UT for the changed behaviour? Something similar to TestSplitWithCache, but start with a given value for evictOnClose, change it for RSes only, trigger a region close, check the cache:
> 
> 1. Set evictOnClose to false:
>    `UTIL.getConfiguration().setBoolean(EVICT_BLOCKS_ON_CLOSE_KEY, false);`
> 2. Start cluster:
>    `UTIL.startMiniCluster(1);`
> 3. Change the config for the RS only:
>    `UTIL.getMiniHBaseCluster().getRegionServer(0).getConfiguration().setBoolean(EVICT_BLOCKS_ON_CLOSE_KEY, true);`
> 4. Disable table:
>    `UTIL.getAdmin().disableTable();`
> 5. Check blocks got evicted:

Added the unit test TestUseRSCacheConfigWhileClosingRegion.java

wchevreuil reviewed (changes requested):
The UT seems correct, but let's refrain from duplicating code whenever possible. Please move these new tests to TestSplitWithCache, and reuse as much of the code from there.

ragarkar commented:
> The UT seems correct, but let's refrain from duplicating code whenever possible. Please move these new tests to TestSplitWithCache, and reuse as much of the code from there.

Refactored the unit test and renamed TestSplitWithCache to TestCacheEviction. The new test can be used to test other eviction scenarios in the future.

Apache-HBase commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 41s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  |  Patch does not have any anti-patterns.  |
|||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 52s |  |  master passed  |
| +1 :green_heart: |  compile  |   3m  0s |  |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 39s |  |  master passed  |
| +1 :green_heart: |  spotbugs  |   1m 34s |  |  master passed  |
| +1 :green_heart: |  spotless  |   0m 46s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| -1 :x: |  mvninstall  |   1m 42s | [/patch-mvninstall-root.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6197/4/artifact/yetus-general-check/output/patch-mvninstall-root.txt) |  root in the patch failed.  |
| -1 :x: |  compile  |   1m 41s | [/patch-compile-hbase-server.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6197/4/artifact/yetus-general-check/output/patch-compile-hbase-server.txt) |  hbase-server in the patch failed.  |
| -0 :warning: |  javac  |   1m 41s | [/patch-compile-hbase-server.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6197/4/artifact/yetus-general-check/output/patch-compile-hbase-server.txt) |  hbase-server in the patch failed.  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   0m 35s |  |  the patch passed  |
| -1 :x: |  spotbugs  |   0m 49s | [/patch-spotbugs-hbase-server.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6197/4/artifact/yetus-general-check/output/patch-spotbugs-hbase-server.txt) |  hbase-server in the patch failed.  |
| -1 :x: |  hadoopcheck  |   2m  1s |  |  The patch causes 16 errors with Hadoop v3.3.6.  |
| -1 :x: |  hadoopcheck  |   4m  3s |  |  The patch causes 16 errors with Hadoop v3.4.0.  |
| +1 :green_heart: |  spotless  |   0m 42s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 10s |  |  The patch does not generate ASF License warnings.  |
|  |   |  20m 34s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6197/4/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6197 |
| Optional Tests | dupname asflicense javac spotbugs checkstyle codespell detsecrets compile hadoopcheck hbaseanti spotless |
| uname | Linux 12d3f4abfab2 5.4.0-186-generic #206-Ubuntu SMP Fri Apr 26 12:31:10 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 87759d04208ab98e649607fb81008775e2653f7f |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| hadoopcheck | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6197/4/artifact/yetus-general-check/output/patch-javac-3.3.6.txt |
| hadoopcheck | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6197/4/artifact/yetus-general-check/output/patch-javac-3.4.0.txt |
| Max. process+thread count | 85 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6197/4/console |
| versions | git=2.34.1 maven=3.9.8 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 45s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 52s |  |  master passed  |
| +1 :green_heart: |  compile  |   1m 11s |  |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 35s |  |  master passed  |
| +1 :green_heart: |  shadedjars  |   6m 50s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| -1 :x: |  mvninstall  |   2m 11s | [/patch-mvninstall-root.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6197/4/artifact/yetus-jdk17-hadoop3-check/output/patch-mvninstall-root.txt) |  root in the patch failed.  |
| -1 :x: |  compile  |   1m 10s | [/patch-compile-hbase-server.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6197/4/artifact/yetus-jdk17-hadoop3-check/output/patch-compile-hbase-server.txt) |  hbase-server in the patch failed.  |
| -0 :warning: |  javac  |   1m 10s | [/patch-compile-hbase-server.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6197/4/artifact/yetus-jdk17-hadoop3-check/output/patch-compile-hbase-server.txt) |  hbase-server in the patch failed.  |
| +1 :green_heart: |  javadoc  |   0m 34s |  |  the patch passed  |
| -1 :x: |  shadedjars  |   5m 20s |  |  patch has 16 errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  |   1m 19s | [/patch-unit-hbase-server.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6197/4/artifact/yetus-jdk17-hadoop3-check/output/patch-unit-hbase-server.txt) |  hbase-server in the patch failed.  |
|  |   |  24m 55s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6197/4/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6197 |
| Optional Tests | javac javadoc unit compile shadedjars |
| uname | Linux adde2e4b6722 5.4.0-192-generic #212-Ubuntu SMP Fri Jul 5 09:47:39 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 87759d04208ab98e649607fb81008775e2653f7f |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| shadedjars | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6197/4/artifact/yetus-jdk17-hadoop3-check/output/patch-shadedjars.txt |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6197/4/testReport/ |
| Max. process+thread count | 83 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6197/4/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 41s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  |  Patch does not have any anti-patterns.  |
|||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  0s |  |  master passed  |
| +1 :green_heart: |  compile  |   2m 58s |  |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 38s |  |  master passed  |
| +1 :green_heart: |  spotbugs  |   1m 34s |  |  master passed  |
| +1 :green_heart: |  spotless  |   0m 44s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 53s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 58s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 58s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   0m 38s |  |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   1m 41s |  |  the patch passed  |
| +1 :green_heart: |  hadoopcheck  |  10m 21s |  |  Patch does not cause any errors with Hadoop 3.3.6 3.4.0.  |
| +1 :green_heart: |  spotless  |   0m 44s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 11s |  |  The patch does not generate ASF License warnings.  |
|  |   |  35m 26s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6197/5/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6197 |
| Optional Tests | dupname asflicense javac spotbugs checkstyle codespell detsecrets compile hadoopcheck hbaseanti spotless |
| uname | Linux ecef815891c3 5.4.0-186-generic #206-Ubuntu SMP Fri Apr 26 12:31:10 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 8955e45f1b290ac2970380f7054a1415634cf2eb |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| Max. process+thread count | 83 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6197/5/console |
| versions | git=2.34.1 maven=3.9.8 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 40s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  2s |  |  master passed  |
| +1 :green_heart: |  compile  |   0m 57s |  |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 31s |  |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 16s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 56s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  0s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   1m  0s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 28s |  |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 15s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 236m 11s |  |  hbase-server in the patch passed.  |
|  |   | 260m 30s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6197/5/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6197 |
| Optional Tests | javac javadoc unit compile shadedjars |
| uname | Linux 9ffbb5282671 5.4.0-186-generic #206-Ubuntu SMP Fri Apr 26 12:31:10 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 8955e45f1b290ac2970380f7054a1415634cf2eb |
| Default Java | Eclipse Adoptium-17.0.11+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6197/5/testReport/ |
| Max. process+thread count | 5388 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6197/5/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.
